ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/MultiLineLoopRuleTest.kt

```diff
@@ -0,0 +1,229 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat.Companion.assertThatRule
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Test
+
+class MultiLineLoopRuleTest {
+    private val multiLineLoopRuleAssertThat = assertThatRule { MultiLineLoopRule() }
+
+    @Test
+    fun `Given loop statements with curly braces on single line`() {
+        val code =
+            """
+            fun foo() {
+                for (i in 1..10) { return 0 }
+                while (true) { return 0 }
+                do { return 0 } while (true)
+            }
+            """.trimIndent()
+        multiLineLoopRuleAssertThat(code).hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given loop statements without curly braces on single line`() {
+        val code =
+            """
+            fun foo() {
+                for (i in 1..10) return 0
+                while (true) return 0
+                do return 0 while (true)
```

Comment:
According to Android style guide:
> Braces are otherwise required for any if, for, when branch, do, and while statements and expressions, even when the body is empty or contains only a single statement.

Kotlin style guide does not mention it explicitly, but I am fine with imposing this on code styles.